kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/metrics/SnapshotReportSuite.scala

```diff
@@ -169,106 +185,141 @@ class SnapshotReportSuite extends AnyFunSuite with MetricsReportTestUtils {
-        path,
-        SnapshotReportExpectations(
-          expectedReportCount = 2,
-          expectException = false,
-          expectedVersion = Optional.of(0),
-          expectedCheckpointVersion = Optional.empty(),
-          expectedProvidedTimestamp = Optional.of(version0timestamp),
-          expectNonEmptyTimestampToVersionResolutionDuration = true))
+  test("SnapshotReport valid queries - no checkpoint") {
+    withTempDir { tempDir =>
+      withSQLConf(DeltaSQLConf.DELTA_WRITE_CHECKSUM_ENABLED.key -> "true") {
+        val path = tempDir.getCanonicalPath
+        // Set up delta table with version 0, 1
+        spark.range(10).write.format("delta").mode("append").save(path)
+        val version0timestamp = System.currentTimeMillis
+        // Since filesystem modification time might be truncated to the second, we sleep to make
+        // sure the next commit is after this timestamp
+        Thread.sleep(1000)
+        spark.range(10).write.format("delta").mode("append").save(path)
+
+        waitForCrcFileToExistElseThrow(path, 0L)
+        waitForCrcFileToExistElseThrow(path, 1L)
+
+        // Test getLatestSnapshot
+        checkSnapshotReport(
+          (engine, path) => tableManager.getSnapshotAtLatest(engine, path),
+          path,
+          SnapshotReportExpectations(
+            expectedReportCount = 1,
+            expectException = false,
+            expectedVersion = Optional.of(1),
+            expectedCheckpointVersion = Optional.empty(),
+            expectedProvidedTimestamp = Optional.empty() // No time travel by timestamp
+          ))
+
+        // Test getSnapshotAsOfVersion
+        checkSnapshotReport(
+          (engine, path) => tableManager.getSnapshotAtVersion(engine, path, 0),
+          path,
+          SnapshotReportExpectations(
+            expectedReportCount = 1,
+            expectException = false,
+            expectedVersion = Optional.of(0),
+            expectedCheckpointVersion = Optional.empty(),
+            expectedProvidedTimestamp = Optional.empty() // No time travel by timestamp
+          ))
+
+        // Test getSnapshotAsOfTimestamp - only if adapter supports it
+        if (tableManager.supportsTimestampResolution) {
+          checkSnapshotReport(
+            (engine, path) => tableManager.getSnapshotAtTimestamp(engine, path, version0timestamp),
+            path,
+            SnapshotReportExpectations(
+              expectedReportCount = 2,
+              expectException = false,
+              expectedVersion = Optional.of(0),
+              expectedCheckpointVersion = Optional.empty(),
+              expectedProvidedTimestamp = Optional.of(version0timestamp),
+              expectNonEmptyTimestampToVersionResolutionDuration = true))
+        }
+      }
     }
   }
 
   test("SnapshotReport valid queries - with checkpoint") {
     withTempDir { tempDir =>
-      val path = tempDir.getCanonicalPath
```

Comment:
all i added was `withSQLConf(DeltaSQLConf.DELTA_WRITE_CHECKSUM_ENABLED.key -> "true") {` and `waitForCrcFileToExistElseThrow(path, 0L)` ---- sorry that github is showing such a huge diff